src/utils/query.js

```diff
@@ -13,21 +13,32 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-import getAPIClient from './api';
 
-export default (endpoint, parameters) =>
-	// prettier-ignore
-	getAPIClient()[endpoint].get(parameters)
-		.then(res => {
-			if (res.data) {
-				if (Array.isArray(res.data)) {
-					if (res.data.length === 0) {
-						throw new Error(`No ${endpoint} found using specified parameters.`);
-					}
-					return res.data[0];
-				}
-				return res.data;
-			}
-			// Get endpoints with 2xx status code should always return with data key.
-			throw new Error('No data was returned.');
-		});
+const handleResponse = (endpoint, res) => {
+	// Get endpoints with 2xx status code should always return with data key.
+	if (!res.data) {
+		throw new Error('No data was returned.');
+	}
+	if (Array.isArray(res.data)) {
+		if (res.data.length === 0) {
+			throw new Error(`No ${endpoint} found using specified parameters.`);
+		}
+		return res.data[0];
+	}
+	return res.data;
+};
+
+export default async (client, endpoint, parameters) => {
+	const queries = Array.isArray(parameters)
```

Comment:
Why not just return directly here instead of storing the return value in a variable?